main.go

```diff
@@ -2,15 +2,29 @@ package main
 
 import (
 	"os"
+	"runtime/debug"
 
 	"github.com/magnusbaeck/logstash-filter-verifier/v2/internal/app"
 )
 
-// GitSummary contains "git describe" output and is automatically
-// populated via linker options when building with govvv.
 var GitSummary = "(unknown)"
 
 func main() {
+	info, _ := debug.ReadBuildInfo()
+	for _, kv := range info.Settings {
+		switch kv.Key {
+		case "vcs":
+			GitSummary = kv.Value + " "
+		case "vcs.revision":
+			GitSummary += kv.Value + " "
+		case "vcs.time":
+			GitSummary += kv.Value + " "
+		case "vcs.modified":
+			if kv.Value == "true" {
+				GitSummary += " dirty"
+			}
+		}
+	}
```

Comment:
Not sure about the details here. Is it guaranteed, that there is a non-empty value, if the key is present? Is the order of the keys guaranteed? Does it make sense to print the complete revision hash or would it be better to truncate it to the first n chars (I think the default for git short format is 7 characters).
To me, it looks like the spaces could get off with this approach (if not dirty, there is an unnecessary space at the end, if dirty, there might be two spaces).

Reply:
Yes, you are right. I've tried to improve the patch, please take another look.